src/pytorch_lightning/utilities/seed.py

```diff
@@ -121,13 +121,19 @@ def pl_worker_init_function(worker_id: int, rank: Optional[int] = None) -> None:
 
 
 def _collect_rng_states() -> Dict[str, Any]:
-    """Collect the global random state of :mod:`torch`, :mod:`numpy` and Python."""
-    return {"torch": torch.get_rng_state(), "numpy": np.random.get_state(), "python": python_get_rng_state()}
+    """Collect the global random state of :mod:`torch`, :mod:`torch.cuda`,:mod:`numpy` and Python."""
+    return {
+        "torch": torch.get_rng_state(),
+        "torch.cuda": torch.cuda.get_rng_state_all(),
+        "numpy": np.random.get_state(),
+        "python": python_get_rng_state(),
+    }
 
 
 def _set_rng_states(rng_state_dict: Dict[str, Any]) -> None:
     """Set the global random state of :mod:`torch`, :mod:`numpy` and Python in the current process."""
     torch.set_rng_state(rng_state_dict["torch"])
+    torch.cuda.set_rng_state_all(rng_state_dict["torch.cuda"])
```

Comment:
I'm wondering, how big is this state (e.g. in bytes)?

Reply:
`sys.getsizeof` reports me 88 bytes

(torch.get_rng_state() -> 72 bytes, and both sizes remain the same after
```
for _ in range(10000000):
        torch.rand(3, 2)
        torch.cuda.FloatTensor(3, 2).normal_()
```

Reply:
Thanks! Looks reasonable, so no problem there :)